front/lib/utils/ampli/index.ts

```diff
@@ -0,0 +1,236 @@
+/* tslint:disable */
+/* eslint-disable */
+// @ts-nocheck
+/**
+ * Ampli - A strong typed wrapper for your Analytics
+ *
+ * This file is generated by Amplitude.
+ * To update run 'ampli pull dust-tt'
+ *
+ * Required dependencies: @amplitude/analytics-node@^1.0.0
+ * Tracking Plan Version: 1
+ * Build: 1.0.0
+ * Runtime: node.js:typescript-ampli-v2
+ *
+ * [View Tracking Plan](https://data.amplitude.com/dust-tt/Dust/events/main/latest)
+ *
+ * [Full Setup Instructions](https://data.amplitude.com/dust-tt/Dust/implementation/dust-tt)
+ */
+
+import * as amplitude from "@amplitude/analytics-node";
+
+export type NodeClient = amplitude.Types.NodeClient;
+export type BaseEvent = amplitude.Types.BaseEvent;
+export type Event = amplitude.Types.Event;
+export type EventOptions = amplitude.Types.EventOptions;
+export type Result = amplitude.Types.Result;
+export type NodeOptions = amplitude.Types.NodeOptions;
+
+export type Environment = "dust";
+
+export const ApiKey: Record<Environment, string> = {
+  dust: "",
+};
+
+/**
+ * Default Amplitude configuration options. Contains tracking plan information.
+ */
+export const DefaultConfiguration: NodeOptions = {
+  plan: {
+    version: "1",
+    branch: "main",
+    source: "dust-tt",
+    versionId: "79fc7267-bc91-42ba-8abc-721cd114724e",
+  },
+  ...{
+    ingestionMetadata: {
+      sourceName: "node.js-typescript-ampli",
+      sourceVersion: "2.0.0",
+    },
+  },
+};
+
+export interface LoadOptionsBase {
+  disabled?: boolean;
+}
+
+export type LoadOptionsWithEnvironment = LoadOptionsBase & {
+  environment: Environment;
+  client?: { configuration?: NodeOptions };
+};
+export type LoadOptionsWithApiKey = LoadOptionsBase & {
+  client: { apiKey: string; configuration?: NodeOptions };
+};
+export type LoadOptionsWithClientInstance = LoadOptionsBase & {
+  client: { instance: NodeClient };
+};
+
+export type LoadOptions =
+  | LoadOptionsWithEnvironment
+  | LoadOptionsWithApiKey
+  | LoadOptionsWithClientInstance;
+
+export interface QuickGuideViewProperties {
+  workspaceId: string;
+  WorkspaceName: string;
+}
+
+export class QuickGuideView implements BaseEvent {
+  event_type = "Quick Guide View";
+
+  constructor(public event_properties: QuickGuideViewProperties) {
+    this.event_properties = event_properties;
+  }
+}
+
+export class SignUp implements BaseEvent {
+  event_type = "Sign Up";
+}
+
+export type PromiseResult<T> = { promise: Promise<T | void> };
+
+const getVoidPromiseResult = () => ({ promise: Promise.resolve() });
+
+// prettier-ignore
+export class Ampli {
+  private disabled: boolean = false;
+  private amplitude?: NodeClient;
+
+  get client(): NodeClient {
+    this.isInitializedAndEnabled();
+    return this.amplitude!;
+  }
+
+  get isLoaded(): boolean {
+    return this.amplitude != null;
+  }
+
+  private isInitializedAndEnabled(): boolean {
+    if (!this.amplitude) {
+      console.error('ERROR: Ampli is not yet initialized. Have you called ampli.load() on app start?');
```

Comment:
Let's use `logger`.

Reply:
Can't use logger here, it's going to be overridden each time we pull from Amplitude (ie: each time we define a new event or property).